### ISSUE TITLE
no-value-for-parameter false positive for mock objects

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#323]
Originally reported by: **Claudiu Popa (BitBucket: [PCManticore](http://bitbucket.org/PCManticore), GitHub: @PCManticore)**

----------------------------------------

This seems to be a little specialised for mocking, but nevertheless the warning shouldn't be reported.

```
#!python

from unittest import mock

class A:
    @mock.patch('sys.modules')
    def test(self, mocked_modules):
        print(mocked_modules)

    def call_test(self):
        self.test() # the warning is triggered here.

a = A()
a.call_test()


```

----------------------------------------
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/323

**Comment by PCManticore** [comment]
Closing this. I'm no longer interested in having this and mock library already does crazy things.
